src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs

```diff
@@ -187,7 +191,41 @@ private IEnumerable<MongoUpdate> ConvertUpdateEntriesToMongoUpdates(IList<IUpdat
 
     private static MongoUpdate ConvertAddedEntryToMongoUpdate(string collectionName, string id, IUpdateEntry entry)
     {
-        throw new NotImplementedException(); // EF-17
+        var entityType = entry.EntityType;
+        var entitySerializer = (IBsonDocumentSerializer)EntitySerializer.Create(entityType);
+
+        var document = new BsonDocument();
+        using (var writer = new BsonDocumentWriter(document))
+        {
+            writer.WriteStartDocument();
+
+            foreach (var property in entityType.GetProperties())
```

Comment:
I'm surprised we can't get the whole instance to serialize, but it is what it is I guess.

Looks like we are forced to serialize the properties one at a time instead of serializing the whole document.

Reply:
The whole document serializer would surely go over the properties as well though but using reflection?

There are some properties that we wouldn't want to serialize on the Clr such as those marked [Ignore] or configured to not be mapped in EF - their GetProperties ensures we only get the properties we have been told to persist.